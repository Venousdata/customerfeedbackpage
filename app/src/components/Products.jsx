// displays all products in a grid format. displayed on home page

import React from "react";
import {
  makeStyles,
  Grid,
  Card,
  CardActionArea,
  CardMedia,
  CardContent,
  Typography,
} from "@material-ui/core";
import { Link } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  gridcontainer: {
    textAlign: "center",
  },
  card: {
    width: 345,
    height: 600,
  },
  media: {
    height: 500,
  },
}));

const Products = ({ products = [] }) => {
  const classes = useStyles();

  return (
    <>
      <Grid
        container
        justify="center"
        spacing={3}
        className={classes.gridcontainer}
        data-testid="products"
      >
        {products.map((product) => (
          <Grid key={product.id} item>
            <Card className={classes.card}>
              <Link to={{ pathname: `/product/${product.id}` }}>
                <CardActionArea>
                  <CardMedia
                    className={classes.media}
                    image={product.image_src}
                    title={product.name}
                  />
                  <CardContent>
                    <Typography gutterBottom variant="h5" component="h2">
                      {product.name}
                    </Typography>
                    <Typography
                      variant="body2"
                      color="textSecondary"
                      component="p"
                    >
                      {product.description}
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Link>
            </Card>
          </Grid>
        ))}
      </Grid>
    </>
  );
};

export default Products;
