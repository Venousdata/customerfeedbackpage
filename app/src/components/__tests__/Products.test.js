import React from "react";
import { render, screen } from "@testing-library/react";
import Products from "../Products";
import { mockProducts } from "./mockData/Products.data";
import { BrowserRouter } from "react-router-dom";

describe("Products tests", () => {
  it("Products should render and match snapshot", (done) => {
    render(
      <BrowserRouter>
        <Products products={mockProducts} />
      </BrowserRouter>
    );
    const component = screen.getByTestId("products");
    expect(component.innerHTML).toMatchSnapshot();
    done();
  });
});
