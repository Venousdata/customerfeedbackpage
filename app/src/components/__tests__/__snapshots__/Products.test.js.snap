// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products tests Products should render and match snapshot 1`] = `"<div class=\\"MuiGrid-root MuiGrid-item\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded\\"><a href=\\"/product/1\\"><button class=\\"MuiButtonBase-root MuiCardActionArea-root\\" tabindex=\\"0\\" type=\\"button\\"><div class=\\"MuiCardMedia-root makeStyles-media-3\\" title=\\"fake name\\"></div><div class=\\"MuiCardContent-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">fake name</h2><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary\\">fake description</p></div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span></button></a></div></div><div class=\\"MuiGrid-root MuiGrid-item\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded\\"><a href=\\"/product/2\\"><button class=\\"MuiButtonBase-root MuiCardActionArea-root\\" tabindex=\\"0\\" type=\\"button\\"><div class=\\"MuiCardMedia-root makeStyles-media-3\\" title=\\"fake namington\\"></div><div class=\\"MuiCardContent-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">fake namington</h2><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary\\">fake description again</p></div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span></button></a></div></div>"`;
